WordPress/gutenberg
# BorderBoxControl: Improve accessibility and labelling

Related:
- https://github.com/WordPress/gutenberg/issues/42095
- https://github.com/WordPress/gutenberg/pull/42348

## What?

Updates the `BorderBoxControl` component to improve labelling, DOM Structure etc.

## Why?

Accessibility is important.

## How?
- Changes the control's wrapping element to a `fieldset` and its primary label to a `legend`
- Updates calls to `getByRole()` in the unit tests to also filter inputs by name.

## Testing Instructions
1. Run `npm run test-unit /packages/components/src/border-box-control/test`
2. Double check the `BorderBoxControl` continues to function correctly within the Storybook
    - http://localhost:50240/?path=/story/components-experimental-borderboxcontrol--default
3. Inspect the storybook control example and confirm the control is wrapped in a `fieldset` with an appropriate `legend` as the label.
4. Edit a post within the block or site editors, add a group block, select it, and then check the border controls still work
